ouroboros-consensus-cardano/src/shelley/Ouroboros/Consensus/Shelley/Ledger/Query.hs

```diff
@@ -434,6 +447,11 @@ instance (ShelleyCompatible proto era, ProtoCrypto proto ~ crypto)
           SL.queryCommitteeMembersState coldCreds hotCreds statuses st
         GetFilteredVoteDelegatees stakeCreds ->
           getFilteredVoteDelegatees st stakeCreds
+        GetPeerSnapshot ->
```

Comment:
This seems relatively inexpensive, but do you know roughly ballpark?

Reply:
I had not analyzed this - as far as calculating the big ledger peers from the whole, it is just some folds and sorting, with strictness applied where appropriate, so it should be well behaved for reasonable input. I'm not sure about the workload of fetching raw peers from the ledger. Should I take a closer overall look?

Reply:
> I'm not sure about the workload of fetching raw peers from the ledger. Should I take a closer overall look?

Perhaps other members of the Networking Team (eg Marcin? Armando?) have already worried about that being "too expensive", since "getting ledger peers" is pretty primary to the Diffusion Layer?

---

We don't need to pin it down exactly. But it'd be regrettable if some SPO's system was performing poorly because they didn't know that they shouldn't be sending this query to often. So we should be able to document the "rough cost" or some such if that sort of thing is a risk.

Reply:
The costs described in https://github.com/IntersectMBO/ouroboros-consensus/pull/1067#issuecomment-2141971151 seem completely fine and in line (or below) what other queries require.